web3c/oasis.js

```diff
@@ -312,6 +474,37 @@ class Oasis {
     };
   }
 
+  _setupInvoke(options = {}) {
+    // for invoke to work, we need to keep track of which subscriptions are currently
+    // available to avoid recreating subscriptions.
+    options.oasis = this;
+    this._invokeSubscription = new InvokeSubscription(options);
+
+    this._executeSend = async (subscription, send, contract, sendArgs) => {
+      const promise = this._invokeSubscription.createResolvablePromise();
+      const address = contract.address ? contract.address : contract.options.address;
+      await send.apply(contract, sendArgs)
+        .on('transactionHash', hash => {
+          this._invokeSubscription.pushExpectedTransaction(subscription.fromAddress, {
+            transactionHash: hash,
+            toAddress: address,
+            promise: promise
+          });
+        });
+
+      return promise;
+    };
+
+    this.invoke = async function(fromAddress, contract, send, ...sendArgs) {
```

Comment:
Nit but what do you think about prefixing this with an `_` since it's not the actual top-level api? I'm worried about adding confusion if people do something like `console.log(web3c.oasis)` and they see this method. Might be confusing without the `_` to signal its not meant to be used.

Reply:
fair enough, that sounds good